apiserver/params/network.go

```diff
@@ -0,0 +1,338 @@
+// the API requests/responses. See also network.Address, from/to
+// which this is transformed.
+type Address struct {
+	Value       string `json:"Value"`
+	Type        string `json:"Type"`
+	NetworkName string `json:"NetworkName"`
+	Scope       string `json:"Scope"`
+}
+
+// FromNetworkAddress is a convenience helper to create a parameter
+// out of the network type, here for Address.
+func FromNetworkAddress(naddr network.Address) Address {
+	return Address{
+		Value:       naddr.Value,
+		Type:        string(naddr.Type),
+		NetworkName: naddr.NetworkName,
+		Scope:       string(naddr.Scope),
+	}
+}
+
+// NetworkAddress is a convenience helper to return the parameter
+// as network type, here for Address.
+func (addr Address) NetworkAddress() network.Address {
+	return network.Address{
+		Value:       addr.Value,
+		Type:        network.AddressType(addr.Type),
+		NetworkName: addr.NetworkName,
+		Scope:       network.Scope(addr.Scope),
+	}
+}
+
+// FromNetworkAddresses is a convenience helper to create a parameter
+// out of the network type, here for a slice of Address.
+func FromNetworkAddresses(naddrs []network.Address) []Address {
+	addrs := make([]Address, len(naddrs))
+	for i, naddr := range naddrs {
+		addrs[i] = FromNetworkAddress(naddr)
+	}
+	return addrs
+}
+
+// NetworkAddresses is a convenience helper to return the parameter
+// as network type, here for a slice of Address.
+func NetworkAddresses(addrs []Address) []network.Address {
+	naddrs := make([]network.Address, len(addrs))
+	for i, addr := range addrs {
+		naddrs[i] = addr.NetworkAddress()
+	}
+	return naddrs
+}
+
+// HostPort associates an address with a port. It's used in
+// the API requests/responses. See also network.HostPort, from/to
+// which this is transformed.
+type HostPort struct {
+	Address
+	Port int `json:"Port"`
+}
+
+// FromNetworkHostPort is a convenience helper to create a parameter
+// out of the network type, here for HostPort.
+func FromNetworkHostPort(nhp network.HostPort) HostPort {
+	return HostPort{FromNetworkAddress(nhp.Address), nhp.Port}
+}
+
+// NetworkHostPort is a convenience helper to return the parameter
+// as network type, here for HostPort.
+func (hp HostPort) NetworkHostPort() network.HostPort {
+	return network.HostPort{hp.Address.NetworkAddress(), hp.Port}
+}
+
+// FromNetworkHostPorts is a helper to create a parameter
+// out of the network type, here for a slice of HostPort.
+func FromNetworkHostPorts(nhps []network.HostPort) []HostPort {
+	hps := make([]HostPort, len(nhps))
+	for i, nhp := range nhps {
+		hps[i] = FromNetworkHostPort(nhp)
+	}
+	return hps
+}
+
+// NetworkHostPorts is a convenience helper to return the parameter
+// as network type, here for a slice of HostPort.
+func NetworkHostPorts(hps []HostPort) []network.HostPort {
+	nhps := make([]network.HostPort, len(hps))
+	for i, hp := range hps {
+		nhps[i] = hp.NetworkHostPort()
+	}
+	return nhps
+}
+
+// FromNetworkHostPortMatrix is a helper to create a parameter
```

Comment:
Rather than "matrix", I'd suggest calling this FromNetworkHostsPorts (and renaming the next one similarly) - it's more common in params and also in some other places in the code.